### PR TITLE
Allow missing source file

### LINKED ISSUE #34: ng extract-i18n fails if the source file does not exist
Hi,

I have a project where the source file, messages.xlf, isn't checked into my repository. After switching to ng-extract-i18n-merge v2 and building in a fresh checkout, I get:

```
$ ng extract-i18n
Running ng-extract-i18n-merge for project client-app
running "extract-i18n" ...
An unhandled exception occurred: ENOENT: no such file or directory, open 'src/locale/messages.xlf'
```

This worked when I tried it with an earlier version, maybe v1.3. It looks like it broke with #27 -- it looks like it needs to catch the missing file exception and fall back to the `extract-i18n`-generated file as the "original" source.

As a workaround I've had to change the builder back to @angular-devkit/build-angular:extract-i18n, generate the source xlf file, check it in, then switch back.

### PR DESCRIPTION
**Pull request by strout**
fixes #34

### CODE REVIEW

**Comment by strout** [comment]
Ah I just missed running `git add`.

File: src/builder.spec.ts [hunk 1]
```diff
@@ -79,12 +79,7 @@ describe('Builder', () => {
     }
 
     test('should fail if extract-i18n fails', async () => {
-        await architectHost.addBuilder('@angular-devkit/build-angular:extract-i18n', createBuilder(() => ({success: false}))); // dummy builder
-        // write dummy file, so that reading before extraction will not fail:
-        await fs.writeFile(MESSAGES_XLF_PATH, '<xliff version="2.0" xmlns="urn:oasis:names:tc:xliff:document:2.0" srcLang="de">\n' +
-            '  <file original="ng.template" id="ngi18n">\n' +
-            '  </file>\n' +
-            '</xliff>', 'utf8');
+        architectHost.addBuilder('@angular-devkit/build-angular:extract-i18n', createBuilder(() => ({success: false}))); // dummy builder
```
Comment (strout) [inline src/builder.spec.ts:82]: My editor was complaining about awaiting a void method. It's also happening in other places in this file, but I don't know how you feel about sweeping changes so I only got this one (and the one in the new test).

File: src/builder.spec.ts [hunk 2]
```diff
@@ -80,11 +80,6 @@ describe('Builder', () => {
 
     test('should fail if extract-i18n fails', async () => {
         await architectHost.addBuilder('@angular-devkit/build-angular:extract-i18n', createBuilder(() => ({success: false}))); // dummy builder
-        // write dummy file, so that reading before extraction will not fail:
```
[the remark this answers is not in the document]
Comment (strout) [reply src/builder.spec.ts:83]: That would be good, added one.

File: src/builder.spec.ts (hunk 1 above)
Comment (daniel-sc) [reply src/builder.spec.ts:82]: Not sure either ;)
Always pro "Boy Scout" - but like separate PRs as well..

**Review by daniel-sc — APPROVED** [review]
Thx! I'll try too cut a release later..

File: src/builder.spec.ts [hunk 3]
```diff
@@ -102,6 +97,40 @@ describe('Builder', () => {
...
+            // The "result" member (of type BuilderOutput) is the next output.
+            const result = await run.result;
+            expect(result.success).toBeTruthy();
+
+            // Stop the builder from running. This stops Architect from keeping
+            // the builder-associated states in memory, since builders keep waiting
+            // to be scheduled.
+            await run.stop();
+        } finally {
+            await fs.rm(MESSAGES_XLF_PATH, {force: true});
```
Comment (daniel-sc) [inline src/builder.spec.ts:129]:
```suggestion
            await fs.rm?.(MESSAGES_XLF_PATH, {force: true});
```
Gracefully handle Node v12..

File: src/builder.spec.ts [hunk 4]
```diff
@@ -102,6 +97,40 @@ describe('Builder', () => {
...
+            await fs.rm(MESSAGES_FR_XLF_PATH, {force: true});
```
Comment (daniel-sc) [inline src/builder.spec.ts:130]:
```suggestion
            await fs.rm?.(MESSAGES_FR_XLF_PATH, {force: true});
```
Node v12

**Review by daniel-sc — APPROVED** [review]

File: src/builder.spec.ts (hunk 3 above)
Comment (strout) [reply src/builder.spec.ts]: Ah thanks

**Comment by tchunwei** [comment]
Any chance to support missing target files?
https://github.com/strout/ng-extract-i18n-merge/blob/a004eabf6d0173a08c6be2d64172a83e038b118f/src/builder.ts#L121

**Pull request merged**